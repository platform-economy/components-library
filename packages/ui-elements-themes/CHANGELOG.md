# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## [0.1.2](https://github.com/relayr/ui-elements/compare/@relayr/ui-elements-themes@0.1.1...@relayr/ui-elements-themes@0.1.2) (2019-08-22)

**Note:** Version bump only for package @relayr/ui-elements-themes






## [0.1.1](https://github.com/relayr/ui-elements/compare/@relayr/ui-elements-themes@0.1.0...@relayr/ui-elements-themes@0.1.1) (2019-08-21)

**Note:** Version bump only for package @relayr/ui-elements-themes






# 0.1.0 (2019-08-02)


### Features

* **ui-elements-basic:** new InputLabel component ([abb6076](https://github.com/relayr/ui-elements/commit/abb6076))
* **ui-elements-layout:** add Card component ([e8db0cf](https://github.com/relayr/ui-elements/commit/e8db0cf))
* **ui-elements-tables:** Table components ([#102](https://github.com/relayr/ui-elements/issues/102)) ([b1b8d6f](https://github.com/relayr/ui-elements/commit/b1b8d6f))
* **ui-elements-themes:** custom font linking system ([8e9f0f0](https://github.com/relayr/ui-elements/commit/8e9f0f0))
