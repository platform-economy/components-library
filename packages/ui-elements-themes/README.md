# ui-elements-themes

## Add to project

via npm:

```bash
npm install --save @relayr/ui-elements-themes
```

or Yarn:
```bash
yarn add @relayr/ui-elements-themes
```
