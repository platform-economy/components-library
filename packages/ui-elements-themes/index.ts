export * from './themes/light';
export * from './themes/dark';
export * from './themes/test';
export * from './themes/_common/paletteDark';
export * from './themes/_common/paletteLight';
export * from './themes/_common/methods';
export * from './themes/_common/fonts';
export * from './themes/_common/withTestTheme';
