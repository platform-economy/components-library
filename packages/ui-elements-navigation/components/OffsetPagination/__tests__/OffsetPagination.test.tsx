import * as React from 'react';
import * as TestRenderer from 'react-test-renderer';
import 'jest-styled-components';
import * as enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { withTestTheme } from '@relayr/ui-elements-themes';
import { OffsetPagination } from '../index';

enzyme.configure({ adapter: new Adapter() });

describe('ui-elements-navigation', () => {
  describe('Components', () => {
    describe('OffsetPagination', () => {
      it('should render first page', () => {
        const tree = TestRenderer
          .create(withTestTheme(<OffsetPagination limit={10} offset={0} totalItems={200} />))
          .toJSON();
        expect(tree).toMatchSnapshot();
      });
      it('should render last page', () => {
        const tree = TestRenderer
          .create(withTestTheme(<OffsetPagination limit={10} offset={190} totalItems={200} />))
          .toJSON();
        expect(tree).toMatchSnapshot();
      });
      it('should render middle page', () => {
        const tree = TestRenderer
          .create(withTestTheme(<OffsetPagination limit={10} offset={100} totalItems={200} />))
          .toJSON();
        expect(tree).toMatchSnapshot();
      });
      it('should render with custom icons', () => {
        const tree = TestRenderer
          .create(withTestTheme(
            <OffsetPagination
              limit={10}
              offset={100}
              totalItems={200}
              previousNavIcon={<div>&lt;</div>}
              nextNavIcon={<div>&gt;</div>}
              firstNavIcon={<div>&lt;&lt;</div>}
              lastNavIcon={<div>&gt;&gt;</div>}
            />))
          .toJSON();
        expect(tree).toMatchSnapshot();
      });
      it('should render with first/last buttons', () => {
        const tree = TestRenderer
          .create(withTestTheme(
            <OffsetPagination
              limit={10}
              offset={100}
              totalItems={200}
              showFirstAndLastNav={true}
              previousNavIcon={<div>&lt;</div>}
              nextNavIcon={<div>&gt;</div>}
              firstNavIcon={<div>&lt;&lt;</div>}
              lastNavIcon={<div>&gt;&gt;</div>}
            />))
          .toJSON();
        expect(tree).toMatchSnapshot();
      });
      it('should call onChange when clicked on page', () => {
        const handler = jest.fn();
        const tree = enzyme
          .mount(withTestTheme(
            <OffsetPagination
              limit={10}
              offset={100}
              totalItems={200}
              siblingRange={2}
              onChange={handler}
            />));
        tree.find('.page-9').simulate('click');
        expect(handler).toHaveBeenCalledTimes(1);
        expect(handler).lastCalledWith({ newOffset: 80, newPageIndex: 8 }, expect.anything());
      });
      it('should NOT call onChange when clicked on current page', () => {
        const handler = jest.fn();
        const tree = enzyme
          .mount(withTestTheme(
            <OffsetPagination
              limit={10}
              offset={100}
              totalItems={200}
              onChange={handler}
            />));
        tree.find('.page-11').simulate('click');
        expect(handler).toHaveBeenCalledTimes(0);
      });
      it('should call onChange when clicked on "previous" button', () => {
        const handler = jest.fn();
        const tree = enzyme
          .mount(withTestTheme(
            <OffsetPagination
              limit={10}
              offset={100}
              totalItems={200}
              onChange={handler}
            />));
        tree.find('.nav-prev').simulate('click');
        expect(handler).toHaveBeenCalledTimes(1);
        expect(handler).lastCalledWith({ newOffset: 90, newPageIndex: 9 }, expect.anything());
      });
      it('should NOT call onChange when clicked on disabled "previous" button', () => {
        const handler = jest.fn();
        const tree = enzyme
          .mount(withTestTheme(
            <OffsetPagination
              limit={10}
              offset={0}
              totalItems={200}
              onChange={handler}
            />));
        tree.find('.nav-prev').simulate('click');
        expect(handler).toHaveBeenCalledTimes(0);
      });
      it('should call onChange when clicked on "next" button', () => {
        const handler = jest.fn();
        const tree = enzyme
          .mount(withTestTheme(
            <OffsetPagination
              limit={10}
              offset={100}
              totalItems={200}
              onChange={handler}
            />));
        tree.find('.nav-next').simulate('click');
        expect(handler).toHaveBeenCalledTimes(1);
        expect(handler).lastCalledWith({ newOffset: 110, newPageIndex: 11 }, expect.anything());
      });
      it('should NOT call onChange when clicked on disabled "next" button', () => {
        const handler = jest.fn();
        const tree = enzyme
          .mount(withTestTheme(
            <OffsetPagination
              limit={10}
              offset={190}
              totalItems={200}
              onChange={handler}
            />));
        tree.find('.nav-next').simulate('click');
        expect(handler).toHaveBeenCalledTimes(0);
      });
      it('should call onChange when clicked on "first" button', () => {
        const handler = jest.fn();
        const tree = enzyme
          .mount(withTestTheme(
            <OffsetPagination
              limit={10}
              offset={100}
              totalItems={200}
              showFirstAndLastNav={true}
              onChange={handler}
            />));
        tree.find('.nav-first').simulate('click');
        expect(handler).toHaveBeenCalledTimes(1);
        expect(handler).lastCalledWith({ newOffset: 0, newPageIndex: 0 }, expect.anything());
      });
      it('should NOT call onChange when clicked on disabled "first" button', () => {
        const handler = jest.fn();
        const tree = enzyme
          .mount(withTestTheme(
            <OffsetPagination
              limit={10}
              offset={0}
              totalItems={200}
              showFirstAndLastNav={true}
              onChange={handler}
            />));
        tree.find('.nav-first').simulate('click');
        expect(handler).toHaveBeenCalledTimes(0);
      });
      it('should call onChange when clicked on "last" button', () => {
        const handler = jest.fn();
        const tree = enzyme
          .mount(withTestTheme(
            <OffsetPagination
              limit={10}
              offset={100}
              totalItems={200}
              showFirstAndLastNav={true}
              onChange={handler}
            />));
        tree.find('.nav-last').simulate('click');
        expect(handler).toHaveBeenCalledTimes(1);
        expect(handler).lastCalledWith({ newOffset: 190, newPageIndex: 19 }, expect.anything());
      });
      it('should NOT call onChange when clicked on disabled "last" button', () => {
        const handler = jest.fn();
        const tree = enzyme
          .mount(withTestTheme(
            <OffsetPagination
              limit={10}
              offset={190}
              totalItems={200}
              showFirstAndLastNav={true}
              onChange={handler}
            />));
        tree.find('.nav-last').simulate('click');
        expect(handler).toHaveBeenCalledTimes(0);
      });
    });
  });
});
