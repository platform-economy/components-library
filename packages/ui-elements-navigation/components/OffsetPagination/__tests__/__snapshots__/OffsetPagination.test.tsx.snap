// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ui-elements-navigation Components OffsetPagination should render first page 1`] = `
.c0 {
  display: inline-block;
  color: #text;
  font-size: 14px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c0 .nav,
.c0 .page {
  display: inline-block;
  width: 3em;
  padding: 3px;
  cursor: pointer;
  box-sizing: border-box;
  border-top: 2px solid transparent;
}

.c0 .nav:hover,
.c0 .page:hover {
  color: #active_hovered;
}

.c0 .nav {
  vertical-align: middle;
}

.c0 .page {
  font-weight: bold;
  text-align: center;
}

.c0 .page.current {
  cursor: default;
  color: #active;
  border-top-color: #active;
}

.c0 .page.current:hover {
  cursor: default;
  color: #active_hovered;
  border-top-color: #active_hovered;
}

.c0 .nav-prev,
.c0 .nav-first {
  text-align: left;
}

.c0 .nav-next,
.c0 .nav-last {
  text-align: right;
}

.c0 .nav.disabled {
  cursor: default;
  color: #inactive;
}

<div
  className="OffsetPagination c0"
>
  <div
    className="nav nav-prev disabled"
  >
    &lt;
  </div>
  <div
    className="page page-1 current"
    onClick={[Function]}
  >
    1
  </div>
  <div
    className="page page-2"
    onClick={[Function]}
  >
    2
  </div>
  <div
    className="page page-3"
    onClick={[Function]}
  >
    3
  </div>
  <div
    className="page page-4"
    onClick={[Function]}
  >
    ...
  </div>
  <div
    className="page page-20"
    onClick={[Function]}
  >
    20
  </div>
  <div
    className="nav nav-next"
    onClick={[Function]}
  >
    &gt;
  </div>
</div>
`;

exports[`ui-elements-navigation Components OffsetPagination should render last page 1`] = `
.c0 {
  display: inline-block;
  color: #text;
  font-size: 14px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c0 .nav,
.c0 .page {
  display: inline-block;
  width: 3em;
  padding: 3px;
  cursor: pointer;
  box-sizing: border-box;
  border-top: 2px solid transparent;
}

.c0 .nav:hover,
.c0 .page:hover {
  color: #active_hovered;
}

.c0 .nav {
  vertical-align: middle;
}

.c0 .page {
  font-weight: bold;
  text-align: center;
}

.c0 .page.current {
  cursor: default;
  color: #active;
  border-top-color: #active;
}

.c0 .page.current:hover {
  cursor: default;
  color: #active_hovered;
  border-top-color: #active_hovered;
}

.c0 .nav-prev,
.c0 .nav-first {
  text-align: left;
}

.c0 .nav-next,
.c0 .nav-last {
  text-align: right;
}

.c0 .nav.disabled {
  cursor: default;
  color: #inactive;
}

<div
  className="OffsetPagination c0"
>
  <div
    className="nav nav-prev"
    onClick={[Function]}
  >
    &lt;
  </div>
  <div
    className="page page-1"
    onClick={[Function]}
  >
    1
  </div>
  <div
    className="page page-17"
    onClick={[Function]}
  >
    ...
  </div>
  <div
    className="page page-18"
    onClick={[Function]}
  >
    18
  </div>
  <div
    className="page page-19"
    onClick={[Function]}
  >
    19
  </div>
  <div
    className="page page-20 current"
    onClick={[Function]}
  >
    20
  </div>
  <div
    className="nav nav-next disabled"
  >
    &gt;
  </div>
</div>
`;

exports[`ui-elements-navigation Components OffsetPagination should render middle page 1`] = `
.c0 {
  display: inline-block;
  color: #text;
  font-size: 14px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c0 .nav,
.c0 .page {
  display: inline-block;
  width: 3em;
  padding: 3px;
  cursor: pointer;
  box-sizing: border-box;
  border-top: 2px solid transparent;
}

.c0 .nav:hover,
.c0 .page:hover {
  color: #active_hovered;
}

.c0 .nav {
  vertical-align: middle;
}

.c0 .page {
  font-weight: bold;
  text-align: center;
}

.c0 .page.current {
  cursor: default;
  color: #active;
  border-top-color: #active;
}

.c0 .page.current:hover {
  cursor: default;
  color: #active_hovered;
  border-top-color: #active_hovered;
}

.c0 .nav-prev,
.c0 .nav-first {
  text-align: left;
}

.c0 .nav-next,
.c0 .nav-last {
  text-align: right;
}

.c0 .nav.disabled {
  cursor: default;
  color: #inactive;
}

<div
  className="OffsetPagination c0"
>
  <div
    className="nav nav-prev"
    onClick={[Function]}
  >
    &lt;
  </div>
  <div
    className="page page-1"
    onClick={[Function]}
  >
    1
  </div>
  <div
    className="page page-8"
    onClick={[Function]}
  >
    ...
  </div>
  <div
    className="page page-9"
    onClick={[Function]}
  >
    9
  </div>
  <div
    className="page page-10"
    onClick={[Function]}
  >
    10
  </div>
  <div
    className="page page-11 current"
    onClick={[Function]}
  >
    11
  </div>
  <div
    className="page page-12"
    onClick={[Function]}
  >
    12
  </div>
  <div
    className="page page-13"
    onClick={[Function]}
  >
    13
  </div>
  <div
    className="page page-14"
    onClick={[Function]}
  >
    ...
  </div>
  <div
    className="page page-20"
    onClick={[Function]}
  >
    20
  </div>
  <div
    className="nav nav-next"
    onClick={[Function]}
  >
    &gt;
  </div>
</div>
`;

exports[`ui-elements-navigation Components OffsetPagination should render with custom icons 1`] = `
.c0 {
  display: inline-block;
  color: #text;
  font-size: 14px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c0 .nav,
.c0 .page {
  display: inline-block;
  width: 3em;
  padding: 3px;
  cursor: pointer;
  box-sizing: border-box;
  border-top: 2px solid transparent;
}

.c0 .nav:hover,
.c0 .page:hover {
  color: #active_hovered;
}

.c0 .nav {
  vertical-align: middle;
}

.c0 .page {
  font-weight: bold;
  text-align: center;
}

.c0 .page.current {
  cursor: default;
  color: #active;
  border-top-color: #active;
}

.c0 .page.current:hover {
  cursor: default;
  color: #active_hovered;
  border-top-color: #active_hovered;
}

.c0 .nav-prev,
.c0 .nav-first {
  text-align: left;
}

.c0 .nav-next,
.c0 .nav-last {
  text-align: right;
}

.c0 .nav.disabled {
  cursor: default;
  color: #inactive;
}

<div
  className="OffsetPagination c0"
>
  <div
    className="nav nav-prev"
    onClick={[Function]}
  >
    <div>
      &lt;
    </div>
  </div>
  <div
    className="page page-1"
    onClick={[Function]}
  >
    1
  </div>
  <div
    className="page page-8"
    onClick={[Function]}
  >
    ...
  </div>
  <div
    className="page page-9"
    onClick={[Function]}
  >
    9
  </div>
  <div
    className="page page-10"
    onClick={[Function]}
  >
    10
  </div>
  <div
    className="page page-11 current"
    onClick={[Function]}
  >
    11
  </div>
  <div
    className="page page-12"
    onClick={[Function]}
  >
    12
  </div>
  <div
    className="page page-13"
    onClick={[Function]}
  >
    13
  </div>
  <div
    className="page page-14"
    onClick={[Function]}
  >
    ...
  </div>
  <div
    className="page page-20"
    onClick={[Function]}
  >
    20
  </div>
  <div
    className="nav nav-next"
    onClick={[Function]}
  >
    <div>
      &gt;
    </div>
  </div>
</div>
`;

exports[`ui-elements-navigation Components OffsetPagination should render with first/last buttons 1`] = `
.c0 {
  display: inline-block;
  color: #text;
  font-size: 14px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c0 .nav,
.c0 .page {
  display: inline-block;
  width: 3em;
  padding: 3px;
  cursor: pointer;
  box-sizing: border-box;
  border-top: 2px solid transparent;
}

.c0 .nav:hover,
.c0 .page:hover {
  color: #active_hovered;
}

.c0 .nav {
  vertical-align: middle;
}

.c0 .page {
  font-weight: bold;
  text-align: center;
}

.c0 .page.current {
  cursor: default;
  color: #active;
  border-top-color: #active;
}

.c0 .page.current:hover {
  cursor: default;
  color: #active_hovered;
  border-top-color: #active_hovered;
}

.c0 .nav-prev,
.c0 .nav-first {
  text-align: left;
}

.c0 .nav-next,
.c0 .nav-last {
  text-align: right;
}

.c0 .nav.disabled {
  cursor: default;
  color: #inactive;
}

<div
  className="OffsetPagination c0"
>
  <div
    className="nav nav-first"
    onClick={[Function]}
  >
    <div>
      &lt;&lt;
    </div>
  </div>
  <div
    className="nav nav-prev"
    onClick={[Function]}
  >
    <div>
      &lt;
    </div>
  </div>
  <div
    className="page page-1"
    onClick={[Function]}
  >
    1
  </div>
  <div
    className="page page-8"
    onClick={[Function]}
  >
    ...
  </div>
  <div
    className="page page-9"
    onClick={[Function]}
  >
    9
  </div>
  <div
    className="page page-10"
    onClick={[Function]}
  >
    10
  </div>
  <div
    className="page page-11 current"
    onClick={[Function]}
  >
    11
  </div>
  <div
    className="page page-12"
    onClick={[Function]}
  >
    12
  </div>
  <div
    className="page page-13"
    onClick={[Function]}
  >
    13
  </div>
  <div
    className="page page-14"
    onClick={[Function]}
  >
    ...
  </div>
  <div
    className="page page-20"
    onClick={[Function]}
  >
    20
  </div>
  <div
    className="nav nav-next"
    onClick={[Function]}
  >
    <div>
      &gt;
    </div>
  </div>
  <div
    className="nav nav-last"
    onClick={[Function]}
  >
    <div>
      &gt;&gt;
    </div>
  </div>
</div>
`;
